tracer/src/Datadog.Trace/AppSec/InstrumentationGateway.cs

```diff
@@ -95,10 +103,20 @@ private void RaiseEvent(HttpContext context, Span relatedSpan, Func<IDictionary<
                 LogAddressIfDebugEnabled(eventData);
                 eventHandler.Invoke(this, new InstrumentationGatewaySecurityEventArgs(eventData, transport, relatedSpan, eraseExistingAddress));
             }
-            catch (Exception ex)
+            catch (Exception ex) when (ex is not BlockException)
             {
                 Log.Error(ex, "DDAS-0004-00: AppSec failed to process request.");
             }
         }
+
+        internal void RaiseBlockingOpportunity(HttpContext context, Scope scope, ImmutableTracerSettings tracerSettings, Action<InstrumentationGatewayBlockingEventArgs> doBeforeActualBlocking = null)
+        {
+            if (BlockingOpportunity == null)
+            {
+                return;
+            }
+
+            BlockingOpportunity.Invoke(this, new InstrumentationGatewayBlockingEventArgs(context, scope, tracerSettings, doBeforeActualBlocking));
```

Comment:
I think this is the same thing:

```suggestion
            BlockingOpportunity?.Invoke(this, new InstrumentationGatewayBlockingEventArgs(context, scope, tracerSettings, doBeforeActualBlocking));
```

Reply:
yes, fixed in 6f5ca731cbee8cfa6f4544e3c705fd37d48d338f